src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -168,5 +148,42 @@ private BoundExpression LiftRangeExpression(BoundRangeExpression node, Immutable
                 value: conditionalExpression,
                 type: node.Type);
         }
+
+        private BoundExpression MakeRangeExpression(
+            MethodSymbol constructionMethod,
+            BoundExpression left,
+            BoundExpression right)
+        {
+            var F = _factory;
+            switch (constructionMethod.MethodKind)
+            {
+                case MethodKind.Constructor:
+                    // The constructor can always be used to construct a range, but if
+                    // any of the arguments are missing then we need to construct replacement
+                    // Indexes
+                    left = left ?? newIndexZero(fromEnd: false);
+                    right = right ?? newIndexZero(fromEnd: true);
+
+                    return F.New(constructionMethod, ImmutableArray.Create(left, right));
+
+                case MethodKind.Ordinary:
```

Comment:
>case MethodKind.Ordinary: [](start = 16, length = 25)

comments for these different cases would be useful to illustrate scenarios

Reply:
Thanks!

---
In reply to: [260988057](https://github.com/dotnet/roslyn/pull/33679#discussion_r260988057) [](ancestors = 260988057)